pkg/ruler/rules/compat.go

```diff
@@ -48,21 +48,26 @@ func FromProto(rg *RuleGroupDesc) rulefmt.RuleGroup {
 	}
 
 	for i, rl := range rg.GetRules() {
-		recordNode := yaml.Node{}
-		recordNode.SetString(rl.GetRecord())
-		alertNode := yaml.Node{}
-		alertNode.SetString(rl.GetAlert())
 		exprNode := yaml.Node{}
 		exprNode.SetString(rl.GetExpr())
+
 		newRule := rulefmt.RuleNode{
-			Record:      recordNode,
-			Alert:       alertNode,
 			Expr:        exprNode,
 			Labels:      client.FromLabelAdaptersToLabels(rl.Labels).Map(),
 			Annotations: client.FromLabelAdaptersToLabels(rl.Annotations).Map(),
 			For:         model.Duration(rl.GetFor()),
 		}
 
+		if rl.GetRecord() != "" {
```

Comment:
Unrelated to the deletion of namespace but noticed this small bug that's been here for a while - only one of `alert` or `record` should be set.

Reply:
Can we add a test showing the bug and fix?

Reply:
Line 204 on `pkg/ruler/api_test.go` is the assertion that was missing.

Reply:
Ah, missed that. Thanks.